main/main.go

```diff
@@ -270,153 +380,239 @@ func generateDataFiles() error {
 	timer := time.NewTimer(time.Duration(duration) * time.Second)
 
 	select {
 	case err = <-errChan:
-		fmt.Printf("Stop diff generation due to error from dcp client %v\n", err)
+		difftool.logger.Errorf("Stop diff generation due to error from dcp client %v\n", err)
 	case <-timer.C:
-		fmt.Printf("Stop diff generation after specified processing duration\n")
+		difftool.logger.Infof("Stop diff generation after specified processing duration\n")
 	}
 
 	err1 := sourceDcpDriver.Stop()
 	if err1 != nil {
-		fmt.Printf("Error stopping source dcp client. err=%v\n", err1)
+		difftool.logger.Errorf("Error stopping source dcp client. err=%v\n", err1)
 	}
 
 	time.Sleep(delayDurationBetweenSourceAndTarget)
 
 	err1 = targetDcpDriver.Stop()
 	if err1 != nil {
-		fmt.Printf("Error stopping target dcp client. err=%v\n", err1)
+		difftool.logger.Errorf("Error stopping target dcp client. err=%v\n", err1)
 	}
 
 	return err
 }
+
+func (difftool *xdcrDiffTool) retrieveReplicationSpecInfo() error {
+	// CBAUTH has already been setup
+	rcMap, err := difftool.remoteClusterSvc.RemoteClusters()
+	if err != nil {
+		difftool.logger.Errorf("Error retrieving remote clusters: %v\n", err)
+		return err
+	}
+
+	specMap, err := difftool.replicationSpecSvc.AllReplicationSpecs()
+	if err != nil {
+		difftool.logger.Errorf("Error retrieving specs: %v\n", err)
+	}
+
+	for _, ref := range rcMap {
+		if ref.Name_ == options.remoteClusterName {
+			difftool.specifiedRef = ref
+			break
+		}
+	}
+
+	for _, spec := range specMap {
+		if spec.SourceBucketName == options.sourceBucketName && spec.TargetBucketName == options.targetBucketName {
+			difftool.specifiedSpec = spec
+			break
+		}
+	}
+
+	var errStrs []string
+	if difftool.specifiedRef == nil {
+		errStrs = append(errStrs, fmt.Sprintf("Unable to find Remote cluster %v\n", options.remoteClusterName))
+	}
+	if difftool.specifiedSpec == nil {
+		errStrs = append(errStrs, fmt.Sprintf("Unable to find Replication Spec with source %v target %v\n", options.sourceBucketName, options.targetBucketName))
+	}
+	if len(errStrs) > 0 {
+		err := fmt.Errorf(strings.Join(errStrs, " and "))
+		difftool.logger.Errorf(err.Error())
+		return err
+	}
+
+	difftool.logger.Infof("Found Remote Cluster: %v and Replication Spec: %v\n", difftool.specifiedRef.String(), difftool.specifiedSpec.String())
+
+	return nil
+}
+
+func (difftool *xdcrDiffTool) populateTemporarySpecAndRef() {
+	difftool.specifiedSpec, _ = metadata.NewReplicationSpecification(options.sourceBucketName, "", /*sourceBucketUUID*/
+		"" /*targetClusterUUID*/, options.targetBucketName, "" /*targetBucketUUID*/)
+
+	difftool.specifiedRef, _ = metadata.NewRemoteClusterReference("" /*uuid*/, "" /*name*/, options.targetUrl, options.targetUsername, options.targetPassword,
+		false /*demandEncryption*/, "" /*encryptionType*/, nil, nil, nil)
+}
+
+func (difftool *xdcrDiffTool) monitorInterruptSignal() {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+	go func() {
```

Comment:
not necessary for yet another go routine?

Reply:
done